.github/workflows/e2e_tests.yaml

```diff
@@ -0,0 +1,182 @@
+name: e2e
+
+on:
+  pull_request:
+    branches:
+      - main
+      - 'release-*'
+    paths-ignore:
+      - 'docs/**'
+      - '**.adoc'
+      - '**.md'
+      - 'LICENSE'
+  push:
+    branches:
+      - main
+      - 'release-*'
+    paths-ignore:
+      - 'docs/**'
+      - '**.adoc'
+      - '**.md'
+      - 'LICENSE'
+
+concurrency:
+  group: ${{ github.head_ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  kubernetes:
+
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Cleanup
+        run: |
+          ls -lart
+          echo "Initial status:"
+          df -h
+
+          echo "Cleaning up resources:"
+          sudo swapoff -a
+          sudo rm -f /swapfile
+          sudo apt clean
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf /opt/ghc
+          sudo rm -rf "/usr/local/share/boost"
+          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+          docker rmi $(docker image ls -aq)
+
+          echo "Final status:"
+          df -h
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Init directories
+        run: |
+          TEMP_DIR="$(pwd)/tmp"
+          mkdir -p "${TEMP_DIR}"
+          echo "TEMP_DIR=${TEMP_DIR}" >> $GITHUB_ENV
+
+          mkdir -p "$(pwd)/bin"
+          echo "$(pwd)/bin" >> $GITHUB_PATH
+
+      - name: Set Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: v1.18
+
+      - name: Container image registry
+        run: |
+          podman run -d --restart=always -p 5001:5000 --name registry registry:2.8.1
+
+          export REGISTRY_ADDRESS=$(hostname -i):5001
```

Comment:
Maybe "localhost" or "127.0.0.1" instead of hostname command will also work. I think that on some different machine it may return multiple IP addresses in the output which may break the functionality.

Reply:
I remember I tried it and it failed. If I remember it correctly, the root cause is connected to IPv6 support in Kubernetes / KinD. Somehow, with `localhost` or even `127.0.0.1`, it involves an IPv6 loopback interface, which breaks the pulling of the image in KinD.

I've just retried it, and it indeed fails: https://github.com/project-codeflare/codeflare-operator/actions/runs/5289718123/jobs/9572984628?pr=136#step:9:229

Reply:
Yes, it is true. Local registry will not be visible from Kind cluster under the IP: 127.0.0.1. It is true. Maybe I have one more idea. We can try to get the IP from docker network inspect kind | jq -r '.[0].IPAM.Config[0].Gateway' . But it is probably a detail.